Required Boost version is missing

The readme does not state the version of Boost that is required for the package. The `CMakeLists.txt` does not check for any particular version.

https://github.com/ismrmrd/siemens_to_ismrmrd/blob/cebbc01d21b37bdf7be507167ed4ad73288f0302/CMakeLists.txt#L69

I built `siemens_to_ismrmrd` with Boost 1.78 and got the following error, https://github.com/SyneRBI/SIRF-SuperBuild/pull/663#issuecomment-1060420865. 

```
/home/runner/work/SIRF-SuperBuild/SIRF-SuperBuild/build/sources/siemens_to_ismrmrd/XNode.cpp:39:22: error: expected primary-expression before ‘const’
  BOOST_FOREACH(XNode const& cnode, node.children_) {
                      ^~~~~
/home/runner/work/SIRF-SuperBuild/SIRF-SuperBuild/build/sources/siemens_to_ismrmrd/XNode.cpp:39:2: error: ‘BOOST_FOREACH’ was not declared in this scope
  BOOST_FOREACH(XNode const& cnode, node.children_) {
  ^~~~~~~~~~~~~
/home/runner/work/SIRF-SuperBuild/SIRF-SuperBuild/build/sources/siemens_to_ismrmrd/XNode.cpp:39:2: note: suggested alternative: ‘BOOST_FINAL’
  BOOST_FOREACH(XNode const& cnode, node.children_) {
  ^~~~~~~~~~~~~
  BOOST_FINAL
/home/runner/work/SIRF-SuperBuild/SIRF-SuperBuild/build/sources/siemens_to_ismrmrd/XNode.cpp: In member function ‘std::__cxx11::string XProtocol::getXMLString::operator()(const XProtocol::XNodeParamMap&) const’:
/home/runner/work/SIRF-SuperBuild/SIRF-SuperBuild/build/sources/siemens_to_ismrmrd/XNode.cpp:98:22: error: expected primary-expression before ‘const’
  BOOST_FOREACH(XNode const& cnode, node.children_) {
                      ^~~~~
/home/runner/work/SIRF-SuperBuild/SIRF-SuperBuild/build/sources/siemens_to_ismrmrd/XNode.cpp:98:2: error: ‘BOOST_FOREACH’ was not declared in this scope
  BOOST_FOREACH(XNode const& cnode, node.children_) {
  ^~~~~~~~~~~~~
/home/runner/work/SIRF-SuperBuild/SIRF-SuperBuild/build/sources/siemens_to_ismrmrd/XNode.cpp:98:2: note: suggested alternative: ‘BOOST_FINAL’
  BOOST_FOREACH(XNode const& cnode, node.children_) {
  ^~~~~~~~~~~~~
  BOOST_FINAL
/home/runner/work/SIRF-SuperBuild/SIRF-SuperBuild/build/sources/siemens_to_ismrmrd/XNode.cpp: In member function ‘std::__cxx11::string XProtocol::getXMLString::operator()(const XProtocol::XNodeParamArray&) const’:
/home/runner/work/SIRF-SuperBuild/SIRF-SuperBuild/build/sources/siemens_to_ismrmrd/XNode.cpp:115:22: error: expected primary-expression before ‘const’
  BOOST_FOREACH(XNode const& cnode, node.children_) {
                      ^~~~~
/home/runner/work/SIRF-SuperBuild/SIRF-SuperBuild/build/sources/siemens_to_ismrmrd/XNode.cpp:115:2: error: ‘BOOST_FOREACH’ was not declared in this scope
  BOOST_FOREACH(XNode const& cnode, node.children_) {
  ^~~~~~~~~~~~~
/home/runner/work/SIRF-SuperBuild/SIRF-SuperBuild/build/sources/siemens_to_ismrmrd/XNode.cpp:115:2: note: suggested alternative: ‘BOOST_FINAL’
  BOOST_FOREACH(XNode const& cnode, node.children_) {
  ^~~~~~~~~~~~~
  BOOST_FINAL
```
Downgrading Boost to 1.72 makes it work. I chose 1.72 because I suppose you are using Ubuntu 20.04 (ubuntu-latest) in your pipelines. In this case Boost is [1.71](https://packages.ubuntu.com/focal/libboost-all-dev) but for whatever reason they[ haven't released the files](https://sourceforge.net/projects/boost/files/boost/1.71.0/), and then I chose the closer version.

--- a/XNode.cpp
+++ b/XNode.cpp
@@ -6,6 +6,7 @@
 #include <boost/spirit/include/phoenix_stl.hpp>
 #include <boost/fusion/include/adapt_struct.hpp>
 #include <boost/algorithm/string.hpp>
+#include <boost/foreach.hpp>
 
 #include <iostream>
 #include <stdlib.h>

